prog/mutation.go

```diff
@@ -183,6 +190,55 @@ func (ctx *mutator) mutateArg() bool {
 	return true
 }
 
+// Select a random argument weighted by its priority
+func (ma *mutationArgs) pseudoRandomArgs(r *randGen) (arg Arg, ctx ArgCtx) {
+
+	// map used to store the indexes of the arguments with the same priority
+	m := make(map[float64][]int)
+
+	sum := float64(0)
+	for _, value := range ma.priorities {
+		sum += value
+	}
+
+	i := 0
+	for i < len(ma.priorities) {
+		prob := ma.priorities[i] / sum //normalized priorities
+		m[prob] = append(m[prob], i)
+		i++
+	}
+
+	// choose the index of the argument based on its priority
+	chosenIdx := randomChoice(m, r)
+
+	return ma.args[chosenIdx], ma.ctxes[chosenIdx]
+}
+
+// Generates a random argument index based on prorities associated with each argument.
+func randomChoice(m map[float64][]int, r *randGen) int {
+	var keys []float64
+	for k := range m {
+		keys = append(keys, k)
+	}
+
+	// sort the priorities in ascending order
+	sort.Float64s(keys)
+
+	// generate a random value
+	randVal := r.Float64()
+
+	// select the bucket based on the generated value
+	limit := float64(0)
+	for _, p := range keys {
+		limit += p * float64(len(m[p]))
+		if randVal < limit {
+			return m[p][r.rand(len(m[p]))] // uniform random choice
+		}
+	}
+
+	return -1
```

Comment:
Can this happen? If not I think it's better to panic right away.

Reply:
Only when invalid values (such as NaN) are received. I switched to panic.